File: 2-add-two-numbers/add-two-numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
       ListNode dummy = new ListNode(0);
       ListNode temp =dummy;
       int carry =0;
       while(l1 != null || l2 !=null){
               int sum =0;
        if (l1 != null) sum += l1.val;
        if (l2 != null) sum += l2.val;
        sum += carry;
        temp.next = new ListNode(sum % 10);
        temp = temp.next;
       if (l1 != null) l1 = l1.next;
        if (l2 != null) l2 = l2.next;
        carry = sum/10;
        }
        if (carry > 0) {
    temp.next = new ListNode(carry);
}
     return dummy.next ;

       }
}

    

File: 2-add-two-numbers/add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode(0)  # Helper node
        current = dummy      # Pointer to build result
        carry = 0           # Carry from previous addition
        while l1 or l2 or carry:
          val1 = l1.val if l1 else 0
          val2 = l2.val if l2 else 0
          total = val1+ val2 +carry
          carry = total // 10
          digit = total % 10
          current.next = ListNode(digit)
          current = current.next
          l1 = l1.next if l1 else None
          l2 =l2.next if l2 else None
        return dummy.next

        